overlord/auth/auth.go

```diff
@@ -335,11 +335,11 @@ NextUser:
 type DeviceAssertions interface {
 	// Model returns the device model assertion.
 	Model() (*asserts.Model, error)
-	// Serial returns the device model assertion.
+	// Serial returns the device serial assertion.
 	Serial() (*asserts.Serial, error)
 
-	// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the device serial assertion.
-	DeviceSessionRequest(nonce string) (*asserts.DeviceSessionRequest, *asserts.Serial, error)
+	// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the device serial and model assertions.
+	DeviceSessionRequest(nonce string) (*asserts.DeviceSessionRequest, *asserts.Serial, *asserts.Model, error)
```

Comment:
at some point having a long list of returned values is just as confusing as having a long list of parameters, and I think it should be handled similarly by returning a struct. Personally I think three is already at the limit for this.